src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -126,48 +126,45 @@ private void backup(final JavaFile javaFile) throws IOException {
-
-        // Compile the test without our fix
-        runMvnInstall();
-
-        // Check if we pass in isolation before fix
-        TestPluginPlugin.info("Running victim test with polluter before adding code from cleaner.");
-        final List<String> withPolluter =
-                // This is to handle the case of a santa clause, where a test will fail with no polluter
-                polluterName.isEmpty() ? ListUtil.fromArray(victimMethod.methodName()) :
-                                         ListUtil.fromArray(polluterName, victimMethod.methodName());
-        if (testOrderPasses(withPolluter)) {
-            TestPluginPlugin.error("Test passes with the polluter, but is supposed to fail.");
-            return;
-        }
-
-        // Do our fix
-        TestPluginPlugin.info("Applying code from cleaner and recompiling.");
-        final NodeList<Statement> cleanerStmts = cleanerMethod.body().getStatements();
-
+    // Try applying cleaner statements and see if test passes when run after polluter
+    // Returns true if passes, returns false if not
+    private boolean checkCleanerStmts(final String polluterName,
+                                      final JavaMethod victimMethod,
+                                      final NodeList<Statement> cleanerStmts) throws Exception {
         // Note: this modifies victimMethod, so when we eventually make it delta-debug, we will want
         // to make a copy of the victimMethod somewhere
         victimMethod.prepend(cleanerStmts);
         victimMethod.javaFile().writeAndReloadCompilationUnit();
 
-        // Recompile and check if we pass in isolation after fix
-        runMvnInstall();
-
+        // Rebuild and see if tests run properly
+        try {
+            runMvnInstall();
+        } catch (Exception ex) {
+            TestPluginPlugin.info("Error building the code, passed in cleaner does not work");
+            // Reset the change
+            victimMethod.removeFirstBlock();
+            return false;
+        }
         // TODO: Output to result files rather than stdout
-        TestPluginPlugin.info("Running victim test in isolation with code from cleaner.");
-        final boolean passInIsolationAfterFix = testOrderPasses(withPolluter);
+        TestPluginPlugin.info("Running victim test with code from cleaner.");
+        List<String> tests;
+        if (polluterName.isEmpty()) {
+            tests = ListUtil.fromArray(victimMethod.methodName());
+        } else {
+            tests = ListUtil.fromArray(polluterName, victimMethod.methodName());
+        }
+        boolean passInIsolationAfterFix = testOrderPasses(tests);
         if (!passInIsolationAfterFix) {
             TestPluginPlugin.error("Fix was unsuccessful. Test still fails with polluter.");
         } else {
             TestPluginPlugin.info("Fix was successful! Fixed file:\n" + victimMethod.javaFile().path());
         }
 
+        // Reset the change
+        victimMethod.removeFirstBlock();
```

Comment:
Do we need a `writeAndReloadCompilationUnit()` here? That seems to be what we do whenever we make a change to the method, though not sure if it's actually required.

Reply:
I think this comment may have been placed in the wrong place, maybe you meant on line 265? I always run that command to ensure that the Java file is overwritten given the new changes. I think I had some issues with that in the prior version of the code in the other repository, so made sure to keep calling this method here.